src/polling.ts

```diff
@@ -90,41 +90,60 @@ export class PollingProcessor {
       });
   }
 
-  private async retrieveFlags(): Promise<void> {
-    try {
-      this.log.debug('Fetching flags started');
-      const response = await this.api.getFeatureConfig(
-        this.environment,
-        this.cluster,
-      );
-      this.log.debug('Fetching flags finished');
-      response.data.forEach((fc: FeatureConfig) =>
-        this.repository.setFlag(fc.feature, fc),
-      );
-    } catch (error) {
-      this.log.error('Error loading flags', error);
-      throw error;
+  private async retrieveFlags(retries = 3): Promise<void> {
```

Comment:
`<padantic-alert>`
This will actually only _retry_ twice, but _try_ thrice.
`</padantic-alert>`

Not a problem. But it's worth setting expectations that this will not "retry" 3 times.

Reply:
Flip - good spot. What would you change the name to in this sort of situation? Or would you alter the code so that it did retry 3 times?

Reply:
It's totally a non-issue. But if you want to remove any doubt, you can drop the `re` part and keep it as `tries`.

Reply:
I should have probably changed `postMetricsWithRetries` to `postMetricsWithMultipleAttempts` too... but that just sounds bad. THought the original was ok.